plugins/webhooks/index.js

```diff
@@ -310,6 +311,93 @@ function pullRequestEvent(pluginOptions, request, reply, parsed) {
         .catch(err => reply(boom.wrap(err)));
 }
 
+/**
+ * Check if the pipeline has a triggered job or not
+ * @method  hasTriggeredJob
+ * @param   {Pipeline}  pipeline    The pipeline to check
+ * @param   {String}    startFrom   The trigger name
+ * @returns {Boolean}               True if the pipeline contains the triggered job
+ */
+function hasTriggeredJob(pipeline, startFrom) {
+    const nextJobs = workflowParser.getNextJobs(pipeline.workflowGraph, {
+        trigger: startFrom
+    });
+
+    return nextJobs.length > 0;
+}
+
+/**
+ * Get all pipelines which has triggered job
+ * @method  triggeredPipelines
+ * @param   {PipelineFactory}   pipelineFactory The pipeline factory to get the branch list from
+ * @param   {Object}            scmConfig       Has the token and scmUri to get branches
+ * @param   {String}            branch          The branch which is committed
+ * @returns {Promise}                           Promise that resolves into triggered pipelines
+ */
+function triggeredPipelines(pipelineFactory, scmConfig, branch) {
+    return pipelineFactory.scm.getBranchList(scmConfig)
+        .then((branches) => {
+            const splitUri = scmConfig.scmUri.split(':');
+
+            // only add non pushed branch, because there is possibility the branch is deleted at filter.
+            return branches.filter(b => b.name !== branch).map((b) => {
+                splitUri[2] = b.name;
+
+                return splitUri.join(':');
+            });
+        })
+        .then(scmUris => pipelineFactory.list({ params: { scmUri: scmUris } }))
+        .then(pipelines => pipelines.filter(p => hasTriggeredJob(p, `~commit:${branch}`)))
+        .then((pipelines) => {
+            const scmUri = scmConfig.scmUri;
```

Comment:
probably don't need to declare this variable here if it is only used once

Reply:
You are right, I fixed.